### DISCUSSION TITLE
Issues with compiling the POM file inside sonar-plugin with maven

### DISCUSSION THREAD

**Discussion by Obi2Newbie** [green-code-initiative/ecoCode#214] [Q&A] [answered]
Hello, I'm currently trying to compile the sonar-plugin to get a jar file for SonarQube but each time I'm receiving an error about MojoExecutionException. Is there any way I can get the jar file for the JS plugin for SonarQube?

**Comment by utarwyn** [comment, marked as answer]
Hello! Can you paste the stack trace of the error? It will make it easier to identify the problem

*ecoCode-javascript* has been reworked completly and moved to its dedicated repository recently (see #211).. and has not yet been released, but will be shortly and will come with a bunch of improvements! Until then you can download the latest built plugin in the release 1.3.0 of ecoCode, [available here](https://github.com/green-code-initiative/ecoCode/releases/tag/1.3.0).

**Reply by utarwyn** [reply]
Oh sorry I did not see that you were talking about the "sonar-plugin" of the new ecoCode-javascript repository.

To compile it from sources, you need to follow sections "Requirements" and "Installation" of the [CONTRIBUTING.md](https://github.com/green-code-initiative/ecoCode-javascript/blob/main/CONTRIBUTING.md#requirements) file before. The "MojoExecutionException" may occurs if you do not have installed the necessary Node.js packages with `yarn install`.

**Reply by Obi2Newbie** [reply]
Hi sorry for the unnecessary post yesterday. Today I tried to recompile the file again and it did works. Don't know what happened yesterday for the build to fail multiple times but thanks for the link to the 1.3.0 release of ecoCode. I'll be marking it as the answer.
